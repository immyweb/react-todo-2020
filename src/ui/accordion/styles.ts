import glamorous from 'glamorous';
import { colours } from '../../styles/colours';
import { mediaQueries } from '../../styles/breakpoints';
import { transparentize } from 'polished';

import { H4, H5, A } from '../../ui/typography/typography';

import { ITheme } from '../../types/theme';

// Accordion

export const AccordionContainer = glamorous.section({
  label: 'accordion-container',
  position: 'relative',
  marginBottom: 25,
  background: colours.$white,
  ':after': {
    clear: 'both',
    content: '""',
    display: 'table',
  },
});

export const Border = glamorous.div<{ colour: string }>(
  {
    label: 'border',
    left: 0,
    content: '""',
    position: 'absolute',
    top: 0,
    height: '100%',
    width: 5,
    zIndex: 10,
  },
  ({ colour }) => ({
    background: colour,
  }),
);

// Accordion Item

export const Item = glamorous.div<{
  open: boolean;
  index: number;
  theme: ITheme;
  border?: boolean;
}>(
  {
    label: 'accordion-item',
    position: 'relative',
    overflowY: 'hidden',
    transition: 'border-color 250ms ease',
    ':before': {
      content: '""',
      position: 'absolute',
      height: 0,
      width: 0,
      top: 22,
      left: 5,
      border: '4px solid transparent',
    },
    ':last-child': {
      borderBottom: 0,
    },
  },
  ({ open, index, theme, border }) => ({
    borderLeft: border ? `5px solid ${theme.secondary}` : 'none',
    borderBottom: open
      ? `2px solid ${transparentize(0.8, theme.primary)}`
      : `2px solid ${colours.$white}`,
    label: `${index}--accordion-item`,
  }),
);

// Accordion Item Head

export const Header = glamorous.header<{ open: boolean; colour: string }>(
  {
    label: 'accordion-header',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: '15px 20px',
    minHeight: 80,
    color: colours.$offBlack,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ open, colour }) => ({
    background: open ? colours.$white : transparentize(0.8, colour),
  }),
);

export const BadgeHolder = glamorous.div({
  label: 'badge-holder',
  display: 'flex',
  borderRadius: '50%',
  overflow: 'hidden',
  position: 'relative',
});

export const Badge = glamorous.img({
  label: 'badge',
  width: 50,
  height: 50,
});

export const TextHolder = glamorous.div<{ hasBadge?: boolean }>(
  {
    label: 'title-text',
    flexDirection: 'column',
    position: 'relative',
    paddingRight: 15,
  },
  ({ hasBadge }) => ({
    paddingLeft: hasBadge ? 15 : 0,
    flexBasis: hasBadge ? `calc(100% - 65px)` : '100%',
  }),
);

export const SubTitle = glamorous(H5)({
  label: 'accordion-date',
  marginBottom: 0,
});

export const Title = glamorous(H4)<{ index: number }>(
  {
    label: 'heading',
    marginBottom: 0,
    position: 'relative',
  },
  ({ index }) => ({
    label: `${index}--accordion-h4`,
  }),
);

// Accordion Item Body

export const defaultStyle = {
  transition: `opacity ${150}ms ease-out`,
  opacity: 0,
};

export const transitionStyles = {
  entering: { opacity: 0 },
  entered: { opacity: 1 },
  exiting: { opacity: 1 },
  exited: { opacity: 0 },
};

export const Panel = glamorous.article<{ open: boolean }>(
  {
    label: 'accordion-panel',
    padding: '0 20px',
    overflowY: 'hidden',
    [mediaQueries.tabletPortrait]: {
      borderBottom: 0,
    },
  },
  ({ open }) => ({
    display: open ? 'block' : 'none',
  }),
);

export const Copy = glamorous.div<{ colour: string }>(
  {
    label: 'accordion-copy',
    margin: '0 0 15px',
    '& a': {
      textDecoration: 'none',
    },
  },
  ({ colour }) => ({
    '& a': {
      color: colour,
    },
  }),
);

export const Link = glamorous(A)<{ colour: string }>(
  {
    label: 'accordion-link',
    display: 'inline-block',
  },
  ({ colour }) => ({
    color: colour,
  }),
);
