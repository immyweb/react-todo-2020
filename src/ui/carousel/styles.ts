import glamorous from 'glamorous';

import { colours } from '../../styles/colours';
import { fonts } from '../../styles/fonts';
import { BtnNoStyle } from '../../styles/globals';
import { Direction } from './types';

export const CarouselContainer = glamorous.section({
  label: 'carousel-container',
  paddingBottom: 33,
  marginBottom: 25,
  position: 'relative',
  width: '100%',
});

export const Slide = glamorous.div({
  label: 'carousel-slide',
  cursor: 'pointer',
  padding: '0 10px',
  ':focus': {
    outline: 'none',
  },
});

export const SlideLink = glamorous.a({
  label: 'carousel-link',
  display: 'inline-block',
  width: '100%',
  height: '100%',
  objectFit: 'contain',
  textDecoration: 'none',
});

export const SlideImg = glamorous.img<{ borderColour: string }>(
  {
    label: 'carousel-image',
    width: '100%',
    height: '100%',
    objectFit: 'contain',
  },
  ({ borderColour }) => ({
    borderBottom: `3px solid ${borderColour}`,
  }),
);

export const SlideLabel = glamorous.p<{ txtColour: string }>(
  {
    label: 'carousel-label',
    textAlign: 'center',
    fontFamily: fonts.$theSunHeavyNarrow,
    fontSize: 18,
    paddingTop: 8,
  },
  ({ txtColour }) => ({
    color: txtColour,
  }),
);

export const SlideNav = glamorous(BtnNoStyle)<{ direction: Direction }>(
  {
    label: 'carousel-nav',
    position: 'absolute',
    zIndex: 3,
    bottom: -47,
    width: 48,
    height: 48,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ direction }) => ({
    left: direction === Direction.PREV ? 25 : 'auto',
    right: direction === Direction.NEXT ? 25 : 'auto',
  }),
);

// Pagination

export const PaginationContainer = glamorous.div({
  label: 'pagination-container',
  position: 'absolute',
  bottom: 0,
  zIndex: 2,
  width: '100%',
  height: 20,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
});

export const Dot = glamorous.span<{ colour?: string; label?: string }>(
  {
    display: 'inline-block',
    borderRadius: '50%',
    margin: '0 5px',
  },
  ({ colour, label }) => ({
    label,
    background: colour ? colour : colours.$news,
  }),
);
