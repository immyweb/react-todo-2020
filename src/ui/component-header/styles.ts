import glamorous from 'glamorous';

import { P, H2 } from '../typography/typography';
import { colours } from '../../styles/colours';

import { mediaQueries } from '../../styles/breakpoints';
import { IAlign } from '../../utils/alignment-utils';

interface IGProps {
  inline?: boolean;
  alignment?: IAlign;
  linkColour?: string;
  background?: boolean;
  hasBackground?: boolean;
}

const styles = {
  fontSizeLarge: 28,
  fontSizeSmall: 24,
  inlineHighlight: 16,
  marginBottomLarge: 12,
  h2LineHeight: 1.14,
  pLineHeight: 1.38,
};

export const TitleWrapper = glamorous.div<IGProps>(
  {
    label: 'title-wrapper',
  },
  ({ alignment, background, theme }: any) => ({
    textAlign: alignment.x,
    background: background ? theme.primary : 'none',
    padding: background ? '13px 20px 13px 15px' : 0,
    borderLeftStyle: background ? 'solid' : 'none',
    borderLeftWidth: '5px',
    borderLeftColor: theme.secondary,
  }),
);

export const Title = glamorous(H2)<IGProps>(
  {
    label: 'title',
    fontSize: styles.fontSizeSmall,
    lineHeight: styles.h2LineHeight,
    [mediaQueries.tabletPortrait]: {
      fontSize: styles.fontSizeLarge,
      lineHeight: styles.h2LineHeight,
    },
  },
  ({ inline, alignment, background }: any) => ({
    display: inline ? 'inline-block' : 'block',
    textTransform: background ? 'unset' : inline ? 'unset' : 'uppercase',
    marginBottom: background ? 0 : styles.marginBottomLarge,
    textAlign: alignment.x,
    color: background ? colours.$white : colours.$offBlack,
  }),
);

export const Highlight = glamorous(Title)<IGProps>(
  {
    label: 'highlight',
    marginBottom: 0,
    fontSize: styles.fontSizeSmall,
    [mediaQueries.tabletPortrait]: {
      fontSize: styles.fontSizeLarge,
      marginBottom: 0,
    },
  },
  ({ inline, alignment, fontColour, background }: any) => ({
    fontSize: inline ? styles.fontSizeSmall : styles.inlineHighlight,
    textTransform: 'uppercase',
    textAlign: alignment.x,
    color: fontColour,
    marginRight: inline ? 4 : 0,
    [mediaQueries.tabletPortrait]: {
      fontSize: inline ? styles.fontSizeLarge : styles.inlineHighlight,
    },
  }),
);

export const SubTitle = glamorous(P)<IGProps>(
  {
    label: 'sub-title',
    lineHeight: styles.pLineHeight,
    textTransform: 'lowercase',
    marginBottom: 20,
    [mediaQueries.tabletPortrait]: {
      lineHeight: styles.pLineHeight,
    },
  },
  ({ alignment, linkColour, hasBackground }: any) => ({
    textAlign: alignment.x,
    marginTop: hasBackground ? 20 : 0,
    '& a': {
      color: linkColour,
      textDecoration: 'none',
    },

    '& a:hover': {
      textDecoration: 'underline',
    },
  }),
);
