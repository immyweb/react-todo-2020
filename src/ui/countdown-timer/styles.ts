import glamorous from 'glamorous';
import { H1, H2, P } from '../typography/typography';
import { mediaQueries } from '../../styles/breakpoints';
import { fonts } from '../../styles/fonts';

import { IAlign } from '../../utils/alignment-utils';

export const CountdownWrapper = glamorous.div<{
  backgroundImage?: string;
}>(
  {
    position: 'relative',
    width: '100%',
    height: '100%',
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'cover',
    backgroundPosition: 'center center',
    minHeight: 185,
  },
  ({ backgroundImage }) => ({
    backgroundImage: backgroundImage ? `url(${backgroundImage})` : '',
  }),
);

export const CountdownContent = glamorous.div<{ alignment?: IAlign }>(
  {
    position: 'absolute',
    width: '100%',
    padding: 10,
  },
  ({ alignment }) => ({
    top: alignment ? alignment.y : 0,
    left: alignment ? alignment.x : 0,
    transform: `translate(-${alignment ? alignment.x : 0}, -${alignment ? alignment.y : 0})`,
  }),
);

export const CountdownFrame = glamorous.div({
  minHeight: 80,
  position: 'relative',
});

export const CountdownAligner = glamorous.div<{ alignment?: IAlign }>(
  {
    position: 'absolute',
    width: '100%',
    top: '50%',
  },
  ({ alignment }) => ({
    left: alignment ? alignment.x : 0,
    transform: `translate(-${alignment ? alignment.x : 0}, -50%)`,
  }),
);

export const HeaderWrapper = glamorous.div({
  label: 'header-wrapper',
});

export const Title = glamorous(H2)<{ alignment?: IAlign; textColour: string }>(
  {
    marginBottom: 2,
    textTransform: 'uppercase',
    [mediaQueries.tabletPortrait]: {
      lineHeight: 1,
    },
  },
  ({ alignment, textColour }: any) => ({
    textAlign: alignment ? alignment.x : 'left',
    color: textColour,
  }),
);

export const SubTitle = glamorous(P)<{
  alignment?: IAlign;
  textColour: string;
}>(
  {
    fontSize: 14,
    lineHeight: 1.25,
    marginBottom: 5,
    [mediaQueries.tabletPortrait]: {
      fontSize: 14,
      lineHeight: 1.25,
    },
  },
  ({ alignment, textColour }: any) => ({
    textAlign: alignment ? alignment.x : 'left',
    color: textColour,
  }),
);

export const UnitWrapper = glamorous.div<{
  alignment?: IAlign;
  opacity?: number;
}>(
  {
    transition: `opacity ${500}ms linear`,
    transitionDelay: `${1000}ms`,
    ':nth-child(n) > div': {
      marginRight: 3,
    },
  },
  ({ alignment, opacity }: any) => ({
    opacity: opacity ? opacity : 0,
    textAlign: alignment ? alignment.x : 'left',
  }),
);

export const UnitInstance = glamorous.div({
  display: 'inline-block',
});

export const UnitValue = glamorous(H1)<{ textColour: string }>(
  {
    fontSize: 42,
    lineHeight: 1.17,
    fontFamily: fonts.$theSunHeavyNarrow,
    textAlign: 'center',
    marginBottom: 0,
    minWidth: 47,
    [mediaQueries.tabletPortrait]: {
      fontSize: 48,
      lineHeight: 1.17,
    },
    [mediaQueries.mobile]: {
      minWidth: 62,
      fontSize: 48,
      lineHeight: 1.17,
    },
  },
  ({ textColour }) => ({
    color: textColour,
  }),
);

export const UnitLabel = glamorous(P)<{ textColour: string }>(
  {
    fontSize: 12,
    lineHeight: 1.25,
    marginBottom: 0,
    textAlign: 'center',
    [mediaQueries.tabletPortrait]: {
      fontSize: 14,
      lineHeight: 1.25,
    },
    [mediaQueries.mobile]: {
      fontSize: 14,
      lineHeight: 1.25,
    },
  },
  ({ textColour }) => ({
    color: textColour,
  }),
);

export const DigitContainer = glamorous.div({
  width: 29,
  display: 'inline-block',
  textAlign: 'center',
});
