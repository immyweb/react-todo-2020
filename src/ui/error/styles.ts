import glamorous from 'glamorous';

import { H3 } from '../typography/typography';
import { fonts } from '../../styles/fonts';

interface IGProps {
  backgroundColour?: string;
  marginBottom?: number;
}

export const ErrorWrapper = glamorous.div<IGProps>(
  {
    label: 'error-wrapper',
    position: 'relative',
    width: '100%',
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    minHeight: 200,
  },
  ({ backgroundColour }) => ({
    backgroundColor: backgroundColour,
  }),
);

export const Heading = glamorous(H3)<IGProps>(
  {
    fontFamily: fonts.$theSunHeavyCondensed,
    textAlign: 'center',
    fontSize: 22,
  },
  ({ marginBottom }) => ({
    marginBottom: marginBottom ? marginBottom : 5,
  }),
);

export const Graphic = glamorous.div({
  label: 'graphic',
  position: 'relative',
  margin: '0 auto 10px auto',
});
