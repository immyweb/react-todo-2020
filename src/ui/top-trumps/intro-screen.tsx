import * as React from 'react';

import { TTIntroInner, TTWrapper, TTButton, TTIntroButtonHolder } from './styles';

import { IIntroScreen } from './types';
import { ITheme } from '../../types/theme';

interface IProps extends IIntroScreen {
  className?: string;
  theme: ITheme;
  onStartGame(): void;
}

export const IntroScreen: React.SFC<IProps> = ({
  playButtonText,
  theme,
  onStartGame,
  imageMobile,
  imageDesktop,
}) => {
  return (
    <TTIntroInner
      className={'tt-intro-inner'}
      imageMobile={imageMobile}
      imageDesktop={imageDesktop}
    >
      <TTWrapper>
        <TTIntroButtonHolder className={'tt-intro-button-holder'}>
          <TTButton
            className="tt-intro-button"
            backgroundColour={theme.primary}
            hoverBackgroundColour={theme.secondary}
            onClick={onStartGame}
          >
            {playButtonText}
          </TTButton>
        </TTIntroButtonHolder>
      </TTWrapper>
    </TTIntroInner>
  );
};
