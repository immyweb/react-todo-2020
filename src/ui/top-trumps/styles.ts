import glamorous from 'glamorous';
import { colours } from '../../styles/colours';
import { fonts } from '../../styles/fonts';
import { H3, H2 } from '../typography/typography';
import { mediaQueries } from '../../styles/breakpoints';

export const TTContainer = glamorous.section({
  label: 'tt-container',
  width: '100%',
  paddingTop: '134.9%',
  minHeight: 400,
  minWidth: 280,
  backgroundColor: colours.$black,
  position: 'relative',
  [mediaQueries.mobileInner]: {
    paddingTop: '72.9%',
  },
});

export const TTIntroInner = glamorous.div<{
  imageMobile: string;
  imageDesktop: string;
}>(
  {
    label: 'tt-intro-inner',
    width: '100%',
    height: '100%',
    position: 'absolute',
    top: 0,
    backgroundSize: 'cover',
    backgroundPosition: 'top center',
    backgroundRepeat: 'no-repeat',
  },
  ({ imageMobile, imageDesktop }) => ({
    backgroundImage: `url(${imageMobile})`,
    [mediaQueries.mobileInner]: {
      backgroundImage: `url(${imageDesktop})`,
    },
  }),
);

export const TTDefaultInner = glamorous.div<{
  imageMobile: string;
  imageDesktop: string;
}>(
  {
    label: 'tt-default-inner',
    width: '100%',
    height: '100%',
    position: 'absolute',
    top: 0,
    backgroundSize: 'cover',
    backgroundPosition: 'top center',
    backgroundRepeat: 'no-repeat',
  },
  ({ imageMobile, imageDesktop }) => ({
    backgroundImage: `url(${imageMobile})`,
    [mediaQueries.mobileInner]: {
      backgroundImage: `url(${imageDesktop})`,
    },
  }),
);

export const TTVoteWrapper = glamorous.div<{}>({
  label: 'tt-vote-wrapper',
  display: 'flex',
});

export const TTWrapper = glamorous.section({
  label: 'tt-wrapper',
  margin: '0 5%',
  height: '100%',
  width: '90%',
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  position: 'relative',
  [mediaQueries.mobileInner]: {
    margin: '0 5%',
  },
});

export const TTTopGradient = glamorous.div({
  label: 'ttt-top-gradient',
  width: '100%',
  height: 84,
  background: `linear-gradient(to bottom, ${colours.$offBlack}, transparent)`,
  position: 'absolute',
  top: 0,
  zIndex: 2,
});

export const TTBottomGradient = glamorous.div({
  label: 'ttt-top-gradient',
  width: '100%',
  height: 130,
  background: `linear-gradient(rgba(34, 37, 38, 0.9), rgba(34, 37, 38, 0.9))`,
  position: 'absolute',
  bottom: 0,
  zIndex: 2,
});

export const TTHeadline = glamorous(H3)({
  label: 'ttt-card-first-name',
  fontSize: 18,
  textTransform: 'uppercase',
  textAlign: 'center',
  marginTop: 0,
  fontFamily: fonts.$theSunHeavyCondensed,
  [mediaQueries.mobileInner]: {
    fontSize: 18,
  },
});

export const TTSubHeadline = glamorous(H2)({
  label: 'ttt-card-surname',
  fontSize: 48,
  textTransform: 'uppercase',
  textAlign: 'center',
  [mediaQueries.mobileInner]: {
    fontSize: 46,
  },
});

export const TTButton = glamorous.div<{
  backgroundColour: string;
  hoverBackgroundColour: string;
}>(
  {
    label: 'tt-intro-button',
    display: 'flex',
    margin: 'auto',
    alignItems: 'center',
    justifyContent: 'center',
    fontFamily: fonts.$theSunMedium,
    textAlign: 'center',
    height: 44,
    color: colours.$white,
    [mediaQueries.mobileInner]: {
      width: 250,
    },
  },
  ({ backgroundColour, hoverBackgroundColour }) => ({
    backgroundColor: backgroundColour,
    ':hover': {
      backgroundColor: hoverBackgroundColour,
    },
  }),
);

export const TTIntroButtonHolder = glamorous.div({
  label: 'tt-intro-button-holder',
  position: 'absolute',
  width: '100%',
  bottom: 15,
  cursor: 'pointer',
});

export const TTGame = glamorous.section({
  label: 'tt-game',
});

export const TTCard = glamorous.div<{
  imageMobile: string;
  imageDesktop: string;
}>(
  {
    label: 'tt-card',
    width: '100%',
    height: '91.5%',
    position: 'absolute',
    top: 0,
    backgroundSize: 'cover',
    backgroundPosition: 'top center',
    backgroundRepeat: 'no-repeat',
    [mediaQueries.mobileInner]: {},
  },
  ({ imageMobile, imageDesktop }) => ({
    backgroundImage: `url(${imageMobile})`,
    [mediaQueries.mobileInner]: {
      backgroundImage: `url(${imageDesktop})`,
    },
  }),
);

export const TTProgressBarCircle = glamorous.div<{
  viewed: boolean;
  inactiveColour: string;
}>(
  {
    label: 'tt-progress_bar-circle',
    display: 'inline-block',
    borderRadius: '50%',
  },
  ({ viewed, inactiveColour }) => ({
    backgroundColor: viewed ? colours.$white : inactiveColour,
    width: viewed ? 11 : 7,
    height: viewed ? 11 : 7,
  }),
);

export const TTProgressBarContainer = glamorous.div<{
  bkgnd: string;
  borderColour: string;
}>(
  {
    label: 'tt-progress_bar-container',
    display: 'flex',
    position: 'absolute',
    bottom: 0,
    justifyContent: 'space-between',
    height: '8.5%',
    width: '100%',
    textAlign: 'center',
  },
  ({ bkgnd, borderColour }) => ({
    background: bkgnd,
    borderTop: `1px solid ${borderColour}`,
  }),
);

export const TTProgressBarInnerContainer = glamorous.div({
  label: 'tt-progress_bar-inner-container',
  width: 130,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-evenly',
});

export const TTProgressBarArrowContainer = glamorous.div<{
  right: boolean;
}>(
  {
    label: 'tt-progress_bar-arrow-container',
    padding: 15,
    width: 45,
    display: 'flex',
    cursor: 'pointer',
  },
  ({ right }) => ({
    [right ? 'marginRight' : 'marginLeft']: 15,
  }),
);

export const TTProgressBarArrow = glamorous.div<{
  right: boolean;
  disabled: boolean;
}>(
  {
    label: 'tt-progress_bar-arrow',
  },
  ({ right, disabled }) => ({
    border: `solid ${colours.$white}`,
    opacity: disabled ? 0.2 : 1,
    borderWidth: '0 3px 3px 0',
    display: 'inline-block',
    padding: 5,
    alignSelf: 'center',
    transform: right ? 'rotate(-45deg)' : 'rotate(135deg)',
  }),
);

export const TTVoteContainer = glamorous.div({
  label: 'tt-vote-container',
  height: '100%',
  width: '100%',
  display: 'flex',
  position: 'absolute',
  top: 0,
  backgroundColor: colours.$black,
  flexDirection: 'column',
  alignItems: 'center',
});

export const TTVoteHeadline = glamorous.div<{
  colour: string;
}>(
  {
    label: 'tt-vote-headline',
    fontFamily: fonts.$theSunMedium,
    textTransform: 'uppercase',
    marginTop: 10,
    fontSize: 18,
  },
  ({ colour }) => ({
    color: colour,
  }),
);

export const TTVoteSubHeadline = glamorous.div({
  label: 'tt-vote-sub_headline',
  fontFamily: fonts.$theSunHeavyNarrow,
  color: colours.$white,
  textTransform: 'uppercase',
  fontSize: '38px',
  flex: 2,
  width: '80%',
  textAlign: 'center',
  lineHeight: '40px',
  marginTop: 10,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  [mediaQueries.mobileInner]: {
    flex: 1,
  },
});

export const TTVoteBottomCopy = glamorous.div({
  label: 'tt-vote-bottom_copy',
  fontFamily: fonts.$theSunRegular,
  color: colours.$midGrey,
  fontSize: 16,
  width: '80%',
  flex: 1,
  textAlign: 'center',
  margin: '10px 0',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
});

export const TTVoteButtonContainer = glamorous.div({
  label: 'tt-vote-button_container',
  display: 'flex',
  flexDirection: 'column',
  height: '60%',
  width: '100%',
  cursor: 'pointer',
  marginTop: '38%',
  [mediaQueries.mobileInner]: {
    marginTop: '18%',
  },
});

export const TTVoteButtonWrapper = glamorous.div({
  label: 'tt-vote-button-wrapper',
});

export const TTVoteButton = glamorous.a<{
  colour: string;
  hoverBackgroundColour: string;
}>(
  {
    label: 'tt-vote-button',
    display: 'flex',
    border: 0,
    padding: 0,
    height: '20%',
    marginTop: 10,
    cursor: 'pointer',
    textDecoration: 'none',
  },
  ({ colour, hoverBackgroundColour }) => ({
    backgroundColor: colour,
    ':hover': {
      backgroundColor: hoverBackgroundColour,
    },
  }),
);

export const TTVoteButtonThumbContainer = glamorous.div({
  label: 'tt-vote-button_thumb_container',
  flexBasis: '19%',
  [mediaQueries.mobileInner]: {
    flexBasis: '9%',
  },
});

export const TTVoteButtonThumb = glamorous.img({
  label: 'tt-vote-button_thumb',
  width: 'auto',
  height: '100%',
  display: 'block',
  border: `1px solid ${colours.$darkGrey}`,
});

export const TTVoteButtonText = glamorous.div<{ colour: string }>(
  {
    label: 'tt-vote-button_text',
    fontFamily: fonts.$theSunMedium,
    fontSize: 16,
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignSelf: 'center',
    cursor: 'pointer',
  },
  ({ colour }) => ({
    color: colour,
  }),
);

export const TTResultsButtonLink = glamorous.div<{
  colour: string;
  hoverBackgroundColour: string;
}>(
  {
    label: 'tt-results-button_link',
    fontFamily: fonts.$theSunMedium,
    color: colours.$white,
    position: 'absolute',
    bottom: 15,
    fontSize: 16,
    height: 44,
    maxHeight: 44,
    justifyContent: 'center',
    alignItems: 'center',
    display: 'flex',
    flex: 1,
    width: '100%',
    cursor: 'pointer',
    [mediaQueries.mobileInner]: {
      width: 250,
    },
  },
  ({ colour, hoverBackgroundColour }) => ({
    backgroundColor: colour,
    ':hover': {
      backgroundColor: hoverBackgroundColour,
    },
  }),
);

export const TTResultsChartContainer = glamorous.div({
  label: 'tt-results-chart-container',
  width: '100%',
  height: '60%',
  marginTop: '39%',
  [mediaQueries.mobileInner]: {
    marginTop: '19%',
  },
});
