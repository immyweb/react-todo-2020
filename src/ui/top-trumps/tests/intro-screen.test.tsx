import * as React from 'react';
import { shallow } from 'enzyme';

import { IntroScreen } from '../intro-screen';
import { TTButton } from '../styles';

function shallowRender(props?: any) {
  return shallow(
    <IntroScreen
      class={'youCantBuy'}
      theme={{ primary: '#000000', secondary: '#ffffff' }}
      imageMobile="fightNight.png"
      imageDesktop="fightNightBig.png"
      playButtonText="Friday Night's alright for"
      onStartGame={() => {
        // NOOP
      }}
      {...props}
    />,
  );
}

describe('<IntroScreen />', () => {
  it('renders correctly', () => {
    const wrapper = shallowRender();
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onStartGame when play button is clicked', () => {
    const fn = jest.fn();
    const wrapper = shallowRender({ onStartGame: fn });
    const btn = wrapper.find(TTButton).last();
    btn.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
  });
});
