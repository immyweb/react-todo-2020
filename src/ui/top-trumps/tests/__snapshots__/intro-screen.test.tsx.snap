// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntroScreen /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    class="youCantBuy"
    imageDesktop="fightNightBig.png"
    imageMobile="fightNight.png"
    onStartGame={[Function]}
    playButtonText="Friday Night's alright for"
    theme={
      Object {
        "primary": "#000000",
        "secondary": "#ffffff",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <glamorous(section)>
        <glamorous(div)
          className="tt-intro-button-holder"
        >
          <glamorous(div)
            backgroundColour="#000000"
            className="tt-intro-button"
            hoverBackgroundColour="#ffffff"
            onClick={[Function]}
          >
            Friday Night's alright for
          </glamorous(div)>
        </glamorous(div)>
      </glamorous(section)>,
      "className": "tt-intro-inner",
      "imageDesktop": "fightNightBig.png",
      "imageMobile": "fightNight.png",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <glamorous(div)
          className="tt-intro-button-holder"
        >
          <glamorous(div)
            backgroundColour="#000000"
            className="tt-intro-button"
            hoverBackgroundColour="#ffffff"
            onClick={[Function]}
          >
            Friday Night's alright for
          </glamorous(div)>
        </glamorous(div)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <glamorous(div)
            backgroundColour="#000000"
            className="tt-intro-button"
            hoverBackgroundColour="#ffffff"
            onClick={[Function]}
          >
            Friday Night's alright for
          </glamorous(div)>,
          "className": "tt-intro-button-holder",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "backgroundColour": "#000000",
            "children": "Friday Night's alright for",
            "className": "tt-intro-button",
            "hoverBackgroundColour": "#ffffff",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Friday Night's alright for",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <glamorous(section)>
          <glamorous(div)
            className="tt-intro-button-holder"
          >
            <glamorous(div)
              backgroundColour="#000000"
              className="tt-intro-button"
              hoverBackgroundColour="#ffffff"
              onClick={[Function]}
            >
              Friday Night's alright for
            </glamorous(div)>
          </glamorous(div)>
        </glamorous(section)>,
        "className": "tt-intro-inner",
        "imageDesktop": "fightNightBig.png",
        "imageMobile": "fightNight.png",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <glamorous(div)
            className="tt-intro-button-holder"
          >
            <glamorous(div)
              backgroundColour="#000000"
              className="tt-intro-button"
              hoverBackgroundColour="#ffffff"
              onClick={[Function]}
            >
              Friday Night's alright for
            </glamorous(div)>
          </glamorous(div)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <glamorous(div)
              backgroundColour="#000000"
              className="tt-intro-button"
              hoverBackgroundColour="#ffffff"
              onClick={[Function]}
            >
              Friday Night's alright for
            </glamorous(div)>,
            "className": "tt-intro-button-holder",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "backgroundColour": "#000000",
              "children": "Friday Night's alright for",
              "className": "tt-intro-button",
              "hoverBackgroundColour": "#ffffff",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Friday Night's alright for",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
